dbos/queues_test.go

```diff
@@ -540,8 +540,9 @@ func TestQueueRecovery(t *testing.T) {
 			// Root workflow case
 			result, err := h.GetResult()
 			require.NoError(t, err, "failed to get result from recovered root workflow handle")
-			castedResult, ok := result.([]int)
-			require.True(t, ok, "expected result to be of type []int for root workflow, got %T", result)
+			// Need to re-marshal to []int because JSON serialized it to []interface{}
+			castedResult, err := convertJSONToType[[]int](result)
+			require.NoError(t, err, "failed to convert result to []int")
```

Comment:
Because recovery handles are typeless, we won't be able to resolve that for the user -- that's ok because end-users are not expected to manipulate recovery handles.